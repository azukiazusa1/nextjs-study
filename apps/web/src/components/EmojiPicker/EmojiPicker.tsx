import React from 'react';

import useMessages from '@/hooks/useMessages';

const EmojiPicker = () => {
  const { sendMessage } = useMessages();

  return (
    <div className="btn-group">
      <button className="btn btn-circle" onClick={() => sendMessage('👍')}>
        👍
      </button>
      <button className="btn btn-circle" onClick={() => sendMessage('❤️')}>
        ❤️
      </button>
      <button className="btn btn-circle" onClick={() => sendMessage('😆')}>
        😆
      </button>
      <button className="btn btn-circle" onClick={() => sendMessage('👏')}>
        👏
      </button>
      <button className="btn btn-circle" onClick={() => sendMessage('🦀')}>
        🦀
      </button>
    </div>
  );
};

export default EmojiPicker;
