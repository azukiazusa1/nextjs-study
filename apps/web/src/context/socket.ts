import { createContext } from 'react';
import io, { Socket } from 'socket.io-client';

import config from '@/config';

export const socket = io(config.SOCKET_URL + '/session');
export const SocketContext = createContext<Socket>(undefined as never);
