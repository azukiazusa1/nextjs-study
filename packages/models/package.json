{
  "name": "models",
  "version": "0.0.0",
  "main": "./build/index.js",
  "types": "./build/index.d.ts",
  "license": "MIT",
  "scripts": {
    "lint": "eslint --fix src/**/*.ts",
    "dev": "tsc -w",
    "build": "tsc"
  },
  "dependencies": {
    "eslint-config-custom": "*",
    "eslint": "^8.15.0",
    "typescript": "^4.6.4"
  }
}
