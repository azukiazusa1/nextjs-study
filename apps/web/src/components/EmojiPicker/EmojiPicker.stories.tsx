import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import { SocketContext } from '@/context/socket';

import EmojiPicker from './EmojiPicker';

export default {
  component: EmojiPicker,
} as ComponentMeta<typeof EmojiPicker>;

const dummySocket = {
  emit: () => {},
  on: () => {},
  off: () => {},
} as any;

const Template: ComponentStory<typeof EmojiPicker> = (args) => (
  <SocketContext.Provider value={dummySocket}>
    <EmojiPicker />
  </SocketContext.Provider>
);

export const Default = Template.bind({});
