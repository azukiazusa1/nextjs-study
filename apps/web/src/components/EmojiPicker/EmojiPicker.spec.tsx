import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { REQ_EVENTS } from 'models';
import React from 'react';

import { SocketContext } from '@/context/socket';

import EmojiPicker from './EmojiPicker';

const dummySocket = {
  emit: jest.fn(),
  on: jest.fn(),
  off: jest.fn(),
} as any;

describe('components/EmojiPicker', () => {
  const Template: React.FC = () => (
    <SocketContext.Provider value={dummySocket}>
      <EmojiPicker />
    </SocketContext.Provider>
  );

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    const { container } = render(<Template />);
    expect(container).toMatchSnapshot();
  });

  test('絵文字を選択したら、その絵文字を表示する', async () => {
    render(<Template />);
    const emoji = screen.getByRole('button', { name: '👍' });

    await userEvent.click(emoji);

    expect(dummySocket.emit).toBeCalledWith(REQ_EVENTS.SEND_MESSAGE, '👍');
  });
});
