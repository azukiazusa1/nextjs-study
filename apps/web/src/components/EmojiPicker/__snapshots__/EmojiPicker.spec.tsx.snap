// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EmojiPicker should render 1`] = `
<div>
  <div
    class="btn-group"
  >
    <button
      class="btn btn-circle"
    >
      👍
    </button>
    <button
      class="btn btn-circle"
    >
      ❤️
    </button>
    <button
      class="btn btn-circle"
    >
      😆
    </button>
    <button
      class="btn btn-circle"
    >
      👏
    </button>
    <button
      class="btn btn-circle"
    >
      🦀
    </button>
  </div>
</div>
`;
